### PR TITLE
adding a name and language to supplements, and removing code and note…

### PR DESCRIPTION
**Pull request by jbhouse**
…s from the snippet to be entirely contained within the supplements object. also ran the code base through prettier for formatting. sorry Jay

### CODE REVIEW

File: src/app/models/language.enum.ts [hunk 1]
```diff
@@ -0,0 +1,11 @@
+export enum Language {
```
Comment (jbhouse) [inline src/app/models/language.enum.ts]: this is new

File: src/app/models/snippet.ts [hunk 2]
```diff
@@ -4,47 +4,38 @@ export class Snippet {
   id: string;
```
Comment (jbhouse) [inline src/app/models/snippet.ts]: this has been updated, and will need further updates to enable backwards compatibility

File: src/app/models/supplement.ts [hunk 3]
```diff
@@ -1,9 +1,15 @@
+import { Language } from './language.enum';
```
Comment (jbhouse) [inline src/app/models/supplement.ts]: updated

File: src/app/services/search.service.ts [hunk 4]
```diff
@@ -28,43 +27,52 @@ export class SearchService {
...
+      snippets.sort((a, b) =>
+        a.showing && b.showing ? searchResultsMap.get(b) - searchResultsMap.get(a) : +b.showing - +a.showing
+      );
     } else {
       this.sortService.sortSnippets(snippets);
     }
     this.snippetService.onSearchComplete(snippets);
   }
 
   private isInCode(term: string, snippet: Snippet): boolean {
```
Comment (jbhouse) [inline src/app/services/search.service.ts]: updated, as snippets no longer contain code

File: src/app/components/snippet-box/snippet-box.component.html [hunk 5]
```diff
@@ -1,46 +1,136 @@
 <div class="title-row">
```
Comment (jbhouse) [inline src/app/components/snippet-box/snippet-box.component.html]: this is the largest portion of changes

File: src/app/components/snippet-box/snippet-box.component.html [hunk 6]
```diff
@@ -1,46 +1,136 @@
...
+    name="name"
+    placeholder="Name"
+    i18n-placeholder="@@name"
+    spellcheck="false"
+    i18n-title="The name for this supplement@@name"
+    i18n-name="@@name"
+    i18n-aria-label="@@name"
+  />
+
+  <!-- please add styling to this. I do not understand your flexbox fanciness -->
```
Comment (jbhouse) [inline src/app/components/snippet-box/snippet-box.component.html]: please address this comment :-)

File: src/app/components/snippet-box/snippet-box.component.ts [hunk 7]
```diff
@@ -1,7 +1,6 @@
 import { Component, OnInit, Input, OnDestroy } from '@angular/core';
```
Comment (jbhouse) [inline src/app/components/snippet-box/snippet-box.component.ts]: the other major changes are in here

**Comment by jbhouse** [comment]
codecoffer

File: src/app/directives/filedrag.directive.ts [hunk 8]
```diff
@@ -59,20 +57,18 @@ export class FiledragDirective {
...
         //import CodeCoffer JSON
         const newSnippets: Snippet[] = JSON.parse(read.result.toString());
         this.service.import(newSnippets);
-      }
-      else {
+      } else {
         //import lone snippet
         const newSnippet: Snippet = new Snippet();
         newSnippet.title = droppedFile.name;
         newSnippet.tags = droppedFile.name;
```
Comment (developerjaym) [inline src/app/directives/filedrag.directive.ts]: You could probably grab the file extension here and then use that to populate the Snippet's language.

File: src/app/components/snippet-box/snippet-box.component.html (hunk 6 above)
Comment (developerjaym) [reply src/app/components/snippet-box/snippet-box.component.html]: We also need i18n here.

File: src/app/components/snippet-box/snippet-box.component.html [hunk 9]
```diff
@@ -1,46 +1,136 @@
...
+    title="Snippet Title"
+    name="title"
+    placeholder="Title"
+    i18n-placeholder="@@title"
+    spellcheck="false"
+    i18n-title="The title for this snippet@@title"
+    i18n-name="@@title"
+    i18n-aria-label="@@title"
+  />
+  <input
```
Comment (developerjaym) [inline src/app/components/snippet-box/snippet-box.component.html]: Would you mind removing this input or commenting it out? I've been meaning to do that, but I haven't gotten around to it.

File: src/app/components/snippet-box/snippet-box.component.html [hunk 10]
```diff
@@ -1,46 +1,136 @@
...
+
+  <!-- please add styling to this. I do not understand your flexbox fanciness -->
+  <select
+    aria-label="Language"
+    title="Supplement Code Language"
+    name="Language"
+    (change)="setSupplementLanguage($event.target.value, supplement)"
+    [ngModel]="supplement.language"
+  >
+    <option *ngFor="let language of supportedLanguages" [ngValue]="getLang(language)" [label]="language"></option>
```
Comment (developerjaym) [inline src/app/components/snippet-box/snippet-box.component.html]: Let's discuss this one tomorrow. I think we might be able to simplify this.

**Pull request merged**